### PR TITLE
A snapshot of live fibers on the work stealing thread pool

### PR DESCRIPTION
**Pull request by vasilmkd**

### CODE REVIEW

File: core/jvm/src/main/scala/cats/effect/unsafe/WorkStealingThreadPool.scala [hunk 1]
```diff
@@ -445,6 +445,28 @@ private[effect] final class WorkStealingThreadPool(
     ()
   }
 
+  /**
+   * Returns a snapshot of the fibers currently live on this thread pool.
+   *
+   * @return
+   *   a set of currently live fibers and a mapping of active fibers associated to their carrier
+   *   threads
```
Comment (armanbilge) [inline core/jvm/src/main/scala/cats/effect/unsafe/WorkStealingThreadPool.scala]: Checking my intuition: if a fiber is in the external queue, it appears in the live set but not in the map keys?

Comment (vasilmkd) [reply core/jvm/src/main/scala/cats/effect/unsafe/WorkStealingThreadPool.scala]: Yeah. Is the wording weird? Feel free to suggest something else.

Comment (armanbilge) [reply core/jvm/src/main/scala/cats/effect/unsafe/WorkStealingThreadPool.scala]: Just to throw another idea out there: what if we returned a `Set` representing the external queue, and a `Thread -> Set` map for the rest? At least for me it better matches conceptually the underlying architecture of the pool

Comment (vasilmkd) [reply core/jvm/src/main/scala/cats/effect/unsafe/WorkStealingThreadPool.scala]: That's a great idea. Let's do it.

Comment (armanbilge) [reply core/jvm/src/main/scala/cats/effect/unsafe/WorkStealingThreadPool.scala]: When we print it out, we could even organize by thread: this thread is running this fiber, and has queued these, etc.

Comment (vasilmkd) [reply core/jvm/src/main/scala/cats/effect/unsafe/WorkStealingThreadPool.scala]: Yeah, totally.

File: core/jvm/src/main/scala/cats/effect/unsafe/WorkStealingThreadPool.scala [hunk 2]
```diff
@@ -445,6 +446,37 @@ private[effect] final class WorkStealingThreadPool(
     ()
   }
 
+  /**
+   * Returns a snapshot of the fibers currently live on this thread pool.
+   *
+   * @return
+   *   a pair consisting of the set of fibers on the external thread and a map associating
```
Comment (armanbilge) [inline core/jvm/src/main/scala/cats/effect/unsafe/WorkStealingThreadPool.scala]:
```suggestion
   *   a pair consisting of the set of fibers on the external queue and a map associating
```

**Review by armanbilge — DISMISSED** [review]
👍 assuming it compiles this time 😉

**Review by armanbilge — APPROVED** [review]

**Pull request merged**